frontend/src/components/LineChart/LineChart.js

```diff
@@ -47,12 +55,19 @@ const LineChart = ({ data, title, xAxis, yAxis }) => {
             labelComponent={
               <VictoryTooltip
                 constrainToVisibleArea
-                flyoutComponent={<ChartFlyOut title={title} xAxisLabel={xAxis.title} yAxisLabel={yAxis.title} />}
+                flyoutComponent={
+                  <ChartFlyOut title={title} xAxisLabel={xAxis.title} yAxisLabel={yAxis.title} variant={variant} />
+                }
               />
             }
           />
         }>
-        <VictoryAxis fixLabelOverlap tickFormat={handleXTickFormat} style={xAxis.angled && angledProperty} />
+        <VictoryAxis
+          {...(variant === 'time_years' && { tickCount: 12 })}
```

Comment:
Instead of setting `tickCount: 12`, which makes the x-axis show January 1st at the start and end of the axis, let's set `tickValues: [new Date("2022-01-01"), ..., new Date("2022-12-01")]` so that the axis ticks go consistently from January to December. Same for `<StackChart>`.